### PR TITLE
feat/scripts: support generating PDFs for platforms

### PR DESCRIPTION
**Pull request by kbdharun**
## Changes

As discussed in [the chatroom](https://matrix.to/#/!zXiOpjSkFTvtMpsenJ:gitter.im/$sS4To9n1vUpB5exkWW5Aa6wQ-ymqMhLIsl-i6FEeTwE?via=gitter.im&via=matrix.org&via=one.ems.host), this PR adds a way to generate custom PDFs for specific platforms manually. Currently, we use `python3 render.py <../../pages>` to generate PDFs (`tldr-book.pdf` by default) and there is a `-o` or `--output` flag to give a custom name to the PDF.

This PR adds a new optional flag `-p` or `--platform`, allowing pages to be generated for the specified platform alone. For example, `python3 render.py ../../pages -p android` will generate a PDF file named `tldr-book-android.pdf`.

You can pass multiple platforms as arguments, for example, `python3 render.py ../../pages -p android sunos` will generate a PDF file named `tldr-book-android+sunos.pdf`.

In the case of custom file names, the platform name is just appended behind the filename, for example `python3 render.py ../../pages -p android sunos -o tldr-book.pdf` will generate a file named `tldr-book-android+sunos.pdf`.

## Reference Outputs

```bash
(venv) kbdharun@homeserver:~/Downloads/tldr/scripts/pdf$ python3 render.py ../../pages -p android sunos
Rendered page 1 of the directory android
Rendered page 2 of the directory android
Rendered page 3 of the directory android
Rendered page 4 of the directory android
Rendered page 5 of the directory android
Rendered page 6 of the directory android
Rendered page 7 of the directory android
Rendered page 8 of the directory android
Rendered page 9 of the directory android
Rendered page 10 of the directory android
Rendered page 11 of the directory android
Rendered page 12 of the directory android
Rendered page 13 of the directory android
Rendered page 14 of the directory android
Rendered page 1 of the directory sunos
Rendered page 2 of the directory sunos
Rendered page 3 of the directory sunos
Rendered page 4 of the directory sunos
Rendered page 5 of the directory sunos
Rendered page 6 of the directory sunos
Rendered page 7 of the directory sunos
Rendered page 8 of the directory sunos
Rendered page 9 of the directory sunos

Converting all pages to PDF...

Created tldr-book-android+sunos.pdf in the current directory!
```

```bash
(venv) kbdharun@homeserver:~/Downloads/tldr/scripts/pdf$ python3 render.py ../../pages -p androi
No pages found for platform androi. Skipping.
```

```bash
(venv) kbdharun@homeserver:~/Downloads/tldr/scripts/pdf$ python3 render.py ../../pages -p android -o "tldr-books.pdf"
Rendered page 1 of the directory android
Rendered page 2 of the directory android
Rendered page 3 of the directory android
Rendered page 4 of the directory android
Rendered page 5 of the directory android
Rendered page 6 of the directory android
Rendered page 7 of the directory android
Rendered page 8 of the directory android
Rendered page 9 of the directory android
Rendered page 10 of the directory android
Rendered page 11 of the directory android
Rendered page 12 of the directory android
Rendered page 13 of the directory android
Rendered page 14 of the directory android

Converting all pages to PDF...

Created tldr-books-android.pdf in the current directory!
```

## Reference assets

Assets generated in the above steps:

[tldr-book-android+sunos.pdf](https://github.com/tldr-pages/tldr/files/13076974/tldr-book-android%2Bsunos.pdf)
[tldr-books-android.pdf](https://github.com/tldr-pages/tldr/files/13076983/tldr-books-android.pdf)

## Checklist

- [x] All changes have been tested and verified
- [x] The code has been formatted in black
- [x] README has been updated

### CODE REVIEW

**Review by Magrid0 — APPROVED** [review]
Looks good to me!

File: scripts/pdf/render.py [hunk 1]
```diff
@@ -67,14 +76,23 @@ def main(loc, colorscheme, output_filename):
...
+    if platform:
+        output_filename_with_platform = (
+            f"{output_filename[:-4]}-{'+'.join(platform)}.pdf"
+        )
+
     html += "</body></html>"
 
     # Writing the PDF to disk
+    if not html.count("<h2"):
+        print(f"No pages found for platform {', '.join(platform)}. Skipping.")
```
Comment (MasterOdin) [inline scripts/pdf/render.py:89]: If someone hit this case where they pointed it an invalid directory or whatever without also specifying a platform, they'd hit a runtime exception here where `platform` is not defined.

File: scripts/pdf/render.py [hunk 2]
```diff
@@ -67,14 +76,23 @@ def main(loc, colorscheme, output_filename):
             html += '<p style="page-break-before: always" ></p>'
             print(f"Rendered page {page_number} of the directory {operating_sys}")
 
+    output_filename_with_platform = output_filename
```
Comment (MasterOdin) [inline scripts/pdf/render.py:79]: This feels weird, where if `platform = None`, then the `with_platform` bit here is wrong as there's no `platform` to add.

I'd just rewrite `output_filename` as the parameter itself isn't used after you have `output_filename_with_platform`.

File: scripts/pdf/render.py (hunk 1 above)
Comment (kbdharun) [reply scripts/pdf/render.py:89]: I guess I will remove this check then.

**Review by MasterOdin — APPROVED** [review]

**Pull request merged**